pdf-bygger/src/main/kotlin/no/nav/pensjon/brev/pdfbygger/LetterMarkupModule.kt

```diff
@@ -65,4 +127,10 @@ internal object LetterMarkupModule : SimpleModule() {
                 return p.codec.treeToValue(node, clazz)
             }
         }
+
+
+    private abstract class AbstractDeserializer<T, V : T>(private val v: Class<V>) : JsonDeserializer<T>() {
```

Comment:
For å gjøre denne litt mindre verbose i bruk så kan du legge til:
```
companion object {
   inline fun <T, reified V : T> deserializer() = object : AbstractDeserializer<T, V>(V::class.java) {}
}
```
Et annet alternativ kan være noe alla (evt. kombinert med den over):
```
private inline fun <reified T, reified V : T> SimpleModule.addInterfaceDeserializer() = addDeserializer(T::class.java, object : AbstractDeserializer<T, V>(V::class.java) {})
```

Reply:
Mulig man kan omgå det hele om LetterMarkupImpl klassene overstyrer typene på feltene sine til de respektive Impl-klassene.